sips/sip-011/sip-011-catastrophic-failure-recovery.md

```diff
@@ -0,0 +1,464 @@
+# Preamble
+
+SIP Number: 011
+
+Title: Catastrophic Blockchain Failures and Recovery
+
+Author: Jude Nelson <jude@stacks.org>
+
+Consideration: Governance
+
+Type: Meta
+
+Status: Draft
+
+Created: 8 February 2021
+
+License: BSD 2-Clause
+
+Sign-off:
+
+Discussions-To: https://github.com/stacksgov/sips
+
+# Abstract
+
+This SIP describes the sanctioned procedures for recovering the state of the
+Stacks blockchain in the event of a _catastrophic_ failure.  A failure qualifies
+as a _catastrophic_ failure if and only if there is no conceivable way for the
+correct nodes in the network to make progress and preserve safety without human
+intervention.
+
+This document describes how to identify a catastrophic failure, and how the
+various parties in the Stacks ecosystem should work together to resuscitate the
+blockchain.  Importantly, these procedures seek to minimize the degree of human
+intervention and code change whenever possible.  Blockchains represent a social
+contract between the people who run the nodes (see SIP-000), so care must be
+taken to preserve the blockchain's expected SIP-described rules and behaviors
+above all else.
+
+# License and Copyright
+
+This SIP is made available under the terms of the BSD-2-Clause license,
+available at https://opensource.org/licenses/BSD-2-Clause.  This SIP’s copyright
+is held by the Stacks Open Internet Foundation.
+
+# Introduction
+
+What does it mean for a blockchain to fail?  They are designed to be resilient
+to node churn, individual node crashes, and (short-lived) network partitions,
+but two broad categories of bugs can cause the whole blockchain network to enter
+an irrecoverable state in which human intervention becomes necessary to recover.
+This document qualifies these failure modes, and describes various procedures
+for recovering from them.
+
+The first class of bugs are _liveness failures_.  These are bugs that prevent
+the blockchain from making progress.  For example, a bug that causes all nodes
+in the network to simultaneously crash is a liveness failure.  As another
+example, a bug that causes all nodes in the network to become unable to respond
+to network requests is a liveness failure.
+
+The second class of bugs are _safety failures_.  These are bugs in which the
+blockchain itself permits state-transitions that are undesirable and
+inconsistent with the design of the system.  For example, a bug that would let
+anyone move anyone else's STX tokens would be a safety failure.  As another
+example, a bug in which some Clarity code's execution led to the wrong result
+(but consistent across the network) would be a safety failure.
+
+This document describes the ways in which Stacks ecosystem participants can
+coordinate to recover from various instances of these failure modes, and when it
+is appropriate to use them.  These procedures include:
+
+* Announcing a new point-release of the Foundation's reference implementation of
```

Comment:
[nit] Should we note somewhere at the top that "Foundation" references in this doc specifically mean the "Stacks Foundation"?

Reply:
Addressed in 3832ae2